File: codejam/practice/reverse-words/reverse-words.py
#!/usr/bin/env python
# Steve Phillips / elimisteve
# 2011.04.28

import sys

lines = open(sys.argv[1], 'r').readlines()
#print "\n".join([line.strip() for line in lines])
output_file = open(sys.argv[2], 'w')
answer = []

# it == iteration
gen = (x.strip() for x in lines)
for it in range(int(gen.next())):
    items = [x for x in gen.next().split(' ')]
    pre = "Case #%d: " % (it+1,)
    answer.append(pre + ' '.join(items[::-1]))


output_file.write("\n".join(answer))

File: codejam/practice/reverse-words/reverse-words.go
// Steve Phillips / elimisteve
// 2013.12.28

package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"

	jam "github.com/elimisteve/go.jam"
)

func main() {
	nCases := jam.GetInt()
	r := bufio.NewReader(os.Stdin)
	for i := 0; i < nCases; i++ {
		line, _ := r.ReadString('\n')
		words := strings.Split(line[:len(line)-1], " ")
		var reversed []string
		for _, word := range words {
			reversed = append([]string{word}, reversed...)
		}
		fmt.Printf("Case #%d: %v\n", i+1, strings.Join(reversed, " "))
	}
}
